documentation/internal/Application-And-Review-Flow.md

```diff
@@ -0,0 +1,15 @@
+# Application and Review Flow
+
+TODO add written description
+
+## Overall Diagram
+
+![Application and Review Workflow.png](images/Application-and-Review-Workflow.png)
+
```

Comment:
> useGetReviewStructure renamed to useGetReviewStructureForSection in here:

84ff2eb8ee018cc4755689bcc6c614904e65d1a7

> Also not sure if it should mention the Async version and what is it used for?

https://github.com/openmsupply/application-manager-web-app/issues/729

> Tiny one: route should be application/:id here:

84ff2eb8ee018cc4755689bcc6c614904e65d1a7

> Regarding ApplicationSummaryPage (the file is called ApplicationSummary, but I like ApplicationSummaryPage, nicer to update the file on the repo):

84ff2eb8ee018cc4755689bcc6c614904e65d1a7

> It will redirect to the page with the problem, we agree to not display the Summary with problems:

84ff2eb8ee018cc4755689bcc6c614904e65d1a7

> This one is also done, right? We do trim responses not filled or not changes (on changes_required):

Removed the bit about visibility and added a little bit extra about ON_APPLICATION_SUBMIT

84ff2eb8ee018cc4755689bcc6c614904e65d1a7

> Regarding ApplicationPage, this seems to be a little lost in there : )

I just removed for now, should be done in this issue: https://github.com/openmsupply/application-manager-web-app/issues/729

> Regarding NextButton, it is actually inside component Navigation. Also good to be affirmative at this point so the docs can answer questions we (or others) might have during development:
> Also included in the Navigation component:

Added an issue for this diagram to reflect exactly what's been implemented: https://github.com/openmsupply/application-manager-web-app/issues/729

> Renamed to PageElements and receives as props also isSummary (and instead of isEditable, canEdit)

84ff2eb8ee018cc4755689bcc6c614904e65d1a7

> Perhaps this should have a special docs to described the use of this and SectionWrapper for many pages in the the app : )

Added as part of the same issue above: https://github.com/openmsupply/application-manager-web-app/issues/729

> While you are working on this, might be good to also rename useGetApplicationFullStructure with useGetApplicationStructure which is the latest version of this hook (not important though):
> Maybe this comment can be removed, I think this isn't happening anymore:

84ff2eb8ee018cc4755689bcc6c614904e65d1a7: 

> Couldn't find this one on the code, is it still used? Maybe needs to be renamed (is to the left of useGetApplicationFullStructure block : )):

This is not up to date anymore, revalidation workflow describes it better, added as part of next doc issue, there is a small comment in the diagram that says, For Revalidation See (Revalidation Workflow.drawio), i wonder if that's enough: https://github.com/openmsupply/application-manager-web-app/issues/729
 
> Maybe like it was done for ApplicationWrapper describe main route: /application/:id/review:
>And this should be just the same as the root, we aren't using home:
> Not needed - can be removed (path to it as well):

84ff2eb8ee018cc4755689bcc6c614904e65d1a7

Reply:
👍 

I have added issue #729 as part of [Epic 78](https://github.com/openmsupply/application-manager-web-app/issues/536) so we can work on it as soon as possible - although it seems a bit everywhere. Hopefully everything can be addressed at the same time.